staging/vhost-device-can/src/main.rs

```diff
@@ -0,0 +1,124 @@
+// VIRTIO CAN Emulation via vhost-user
+//
+// Copyright 2023-2024 VIRTUAL OPEN SYSTEMS SAS. All Rights Reserved.
+//          Timos Ampelikiotis <t.ampelikiotis@virtualopensystems.com>
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+mod backend;
+mod can;
+mod vhu_can;
+mod virtio_bindings;
+
+use clap::Parser;
+use log::{error, info};
+use socketcan::{CanSocket, Socket};
+use std::convert::TryFrom;
+use std::path::PathBuf;
+use std::process::exit;
+
+pub(crate) type Result<T> = std::result::Result<T, Error>;
+use crate::backend::{start_backend, Error};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct CanArgs {
+    /// Location of vhost-user Unix domain socket. This is suffixed by 0,1,2..socket_count-1.
+    #[clap(short, long, value_name = "SOCKET")]
+    socket_path: PathBuf,
+
+    /// A can device name to be used for reading (ex. vcan, can0, can1, ... etc.)
+    #[clap(short = 'd', long)]
+    can_devices: String,
+
+    /// Number of guests (sockets) to connect to.
+    #[clap(short = 'c', long, default_value_t = 1)]
+    socket_count: u32,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct VuCanConfig {
+    socket_path: PathBuf,
+    socket_count: u32,
+    can_devices: Vec<String>,
+}
```

Comment:
I think this should be defined in `backend.rs`.

I see backend.rs, as a library that we could use somewhere else to launch a device. So I would move what is needed there. I would leave here only what is needed to handle the CLI.

Reply:
Ok I moved it into the `backend.rs` and kept only the CLI related function in `main.rs`